### PR TITLE
feat: show success screen at end of enrolment from settings [WPB-6792]

### PR DESCRIPTION
**Pull request by theswiftway**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6792" title="WPB-6792" target="_blank">TaskWPB-6792</a>  [iOS]- Show success screen at end of enrolment from settings
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Solutions

show success screen at end of enrolment from settings

Needs releases with:

- [ ] GitHub link to other pull request

#### How to Test

**Steps**
1. Go to Settings -> Devices
2. Tap on the client with Not activated or valid status about to get expired or an expired status
3. Tap on Get Certificate
4. finish enrollment
5. Tap on Details button
Expectation: The chain of certificates should be shown

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by typfel — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift [hunk 1]
```diff
@@ -138,19 +132,20 @@ final class DeviceDetailsViewActionsHandler: DeviceDetailsViewActions, Observabl
     func getProteusFingerPrint() async -> String {
         guard let data = await getProteusFingerprint.invoke(userClient: userClient),
                 let fingerPrint = String(data: data, encoding: .utf8) else {
+            logger.error("Valid fingerprint data is missing", attributes: nil)
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift]:
```suggestion
            logger.error("Valid fingerprint data is missing")
```

File: wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift [hunk 2]
```diff
@@ -59,15 +59,9 @@ final class DeviceDetailsViewActionsHandler: DeviceDetailsViewActions, Observabl
     }
 
     @MainActor
-    func updateCertificate() async throws -> E2eIdentityCertificate? {
-        try await enrollClient()
-    }
-
-    @MainActor
-    func enrollClient() async throws -> E2eIdentityCertificate? {
+    func enrollClient() async throws -> String? {
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift]: question: what's the return type? maybe add what it returns as doc. + does it have to be optional

File: wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift [hunk 3]
```diff
@@ -160,6 +155,7 @@ final class DeviceDetailsViewActionsHandler: DeviceDetailsViewActions, Observabl
         let mlsClientResolver = MLSClientResolver()
         guard let mlsClientID = mlsClientResolver.mlsClientId(for: userClient),
         let mlsGroupId = await fetchSelfConversationMLSGroupID() else {
+            logger.error("MLSGroupID for self was not found", attributes: nil)
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift]:
```suggestion
            logger.error("MLSGroupID for self was not found")
```

**Review by KaterinaWire — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift (hunk 2 above)
Comment (theswiftway) [reply wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift]: Added doc

File: wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift (hunk 1 above)
Comment (theswiftway) [reply wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift]: done

File: wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift (hunk 3 above)
Comment (theswiftway) [reply wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift]: done

**Pull request merged**